### PR TITLE
Remove most layout_top in closure

### PR DESCRIPTION
**Pull request by Ekdohibs**
This removes most uses of layout_top in closure. These changes were split off #1084.

### CODE REVIEW

File: middle_end/closure/closure.ml [hunk 1]
```diff
@@ -977,6 +977,34 @@ let close_approx_var { fenv; cenv } id =
 let close_var env id =
   let (ulam, _app) = close_approx_var env id in ulam
 
+let rec compute_expr_layout kinds lam =
+  match lam with
+  | Lvar id | Lmutvar id -> V.Map.find id kinds
```
Comment (chambart) [inline middle_end/closure/closure.ml:982]: I would prefer to raise something else than `Not_found`
```suggestion
  | Lvar id | Lmutvar id -> begin
      match V.Map.find id kinds with
      | exception Not_found ->
        Misc.fatal_errorf "Unbound layout for variable %a" V.print id
      | k -> k
    end
```

File: middle_end/closure/closure.ml [hunk 2]
```diff
@@ -977,6 +977,34 @@ let close_approx_var { fenv; cenv } id =
...
+    in
+    compute_expr_layout kinds body
+  | Lprim(p, _, _) ->
+    Lambda.primitive_result_layout p
+  | Lswitch(_, _, _, kind) | Lstringswitch(_, _, _, _, kind)
+  | Lstaticcatch(_, _, _, kind) | Ltrywith(_, _, _, kind)
+  | Lifthenelse(_, _, _, kind) | Lregion (_, kind) ->
+    kind
+  | Lstaticraise (_, _) ->
+    Lambda.layout_top (* Need a layout_bottom here *)
```
Comment (chambart) [inline middle_end/closure/closure.ml:1002]: Maybe just declare a layout_bottom in lambda and use it here. Or wait for #1115 which declares it

**Review by chambart — APPROVED** [review]
Ok

**Comment by chambart** [comment]
Also you need to import the addition of `Lamba.layout_letrec` to pass the build

**Review by mshinwell — APPROVED** [review]

**Pull request merged**